core/src/wallet/tezos/TezosLikeAccount.h

```diff
@@ -0,0 +1,161 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+
+#ifndef LEDGER_CORE_TEZOSLIKEACCOUNT_H
+#define LEDGER_CORE_TEZOSLIKEACCOUNT_H
+#include <time.h>
+#include <api/AddressListCallback.hpp>
+#include <api/Address.hpp>
+#include <api/TezosLikeAccount.hpp>
+#include <api/TezosLikeTransactionBuilder.hpp>
+#include <api/StringCallback.hpp>
+#include <api/Event.hpp>
+#include <api/BigIntCallback.hpp>
+#include <wallet/common/AbstractWallet.hpp>
+#include <wallet/common/AbstractAccount.hpp>
+#include <wallet/common/Amount.h>
+#include <wallet/tezos/explorers/TezosLikeBlockchainExplorer.h>
+#include <wallet/tezos/synchronizers/TezosLikeAccountSynchronizer.h>
+#include <wallet/tezos/observers/TezosLikeBlockchainObserver.h>
+#include <wallet/tezos/keychains/TezosLikeKeychain.h>
+#include <wallet/tezos/database/TezosLikeAccountDatabaseEntry.h>
+//#include <wallet/tezos/delegation/TezosLikeOriginatedAccount.h>
+
+namespace ledger {
+    namespace core {
+
+        class TezosOperationQuery : public OperationQuery {
+        public:
+            TezosOperationQuery(const std::shared_ptr<api::QueryFilter>& headFilter,
+                                const std::shared_ptr<DatabaseSessionPool>& pool,
+                                const std::shared_ptr<api::ExecutionContext>& context,
+                                const std::shared_ptr<api::ExecutionContext>& mainContext) : OperationQuery(headFilter, pool, context, mainContext) {
+
+            };
+        protected:
+            virtual soci::rowset<soci::row> performExecute(soci::session &sql) {
+                return _builder.select("o.account_uid, o.uid, o.wallet_uid, o.type, o.date, o.senders, o.recipients,"
+                                               "o.amount, o.fees, o.currency_name, o.trust, b.hash, b.height, b.time, orig_op.uid"
+                        )
+                        .from("operations").to("o")
+                        .outerJoin("blocks AS b", "o.block_uid = b.uid")
+                        .outerJoin("tezos_originated_operations AS orig_op", "o.uid = orig_op.uid")
+                        .execute(sql);
+
+            };
+        };
+
+        class TezosLikeOriginatedAccount;
+        class TezosLikeAccount : public api::TezosLikeAccount, public AbstractAccount {
+        public:
+
+            static const int FLAG_TRANSACTION_IGNORED = 0x00;
+            static const int FLAG_NEW_TRANSACTION = 0x01;
+            static const int FLAG_TRANSACTION_UPDATED = 0x01 << 1;
```

Comment:
I think it’s okay to use `0x02` :D

Reply:
I wanted to group all same flags for different types of operations i.e.  `FLAG_NEW_TRANSACTION = 0x01` , `FLAG_NEW_DELEGATION = 0x02` ... then for updates same i.e. `FLAG_TRANSACTION_UPDATED = 0x10` , `FLAG_DELEGATION_UPDATED = 0x20` ...

Reply:
Ok, but `0x01 << 1` is not `0x10`. 🙂

Reply:
:trollface:  really ? 
My answer above is about rewriting the whole section taking into account all type of transactions with logic above ! Will do it in next update !

Reply:
Finally I used directly the Operation type to have the result of "Update" and only kept one flag for ignored transactions ! 
WDYT @phaazon ? 
And thanks again for the review ! 👍

Reply:
LGTM